packages/vite/src/node/utils.ts

```diff
@@ -87,7 +87,7 @@ export function resolveFrom(
     basedir,
     extensions: ssr ? ssrExtensions : DEFAULT_EXTENSIONS,
     // necessary to work with pnpm
-    preserveSymlinks: preserveSymlinks || isRunningWithYarnPnp || false
+    preserveSymlinks: preserveSymlinks ?? isRunningWithYarnPnp ?? false
```

Comment:
I'm afraid this changes the logic :scream: 
Before it was if `preserveSymlinks` is undefined or false, check next
And then check `isRunningWithYarnPnp`, which could be **true**
But it would return `preserveSymlinks`'s **false** value